FlowCrypt/App/AppContext.swift

```diff
@@ -78,16 +78,25 @@ class AppContext {
         )
     }
 
-    func withSession(_ session: SessionType?) -> AppContext {
-        return AppContext(
-            encryptedStorage: self.encryptedStorage,
+    func withSession(_ session: SessionType?) -> UserContext {
+        guard
+            let authType = dataService.currentAuthType,
+            let user = dataService.currentUser
+        else {
+            fatalError()
+        }
```

Comment:
These should be dependency injected also, I think?. They should be retrieved only once, somewhere all the way up in router or similar place.

Reply:
We have 5 places where we use it: 4 in GlobalRouter and 1 in AppStartup. Also we use `dataService` and the only object that have it is context itself.

Reply:
Sure, but I don't want to have this hidden in here. I'd rather that we do this in `GlobalRouter` and then we refactor it there in the future once we get rid of `dataService`.

It's an eye sore, so putting it in GlobalRouter where it's more visible will make us refactor it later :smirk: